Announce improved SELinux volume relabelling beta (#49919)
kubernetes/website: content/en/docs/tasks/configure-pod-container/security-context.md

{{< note >}}
To assign SELinux labels, the SELinux security module must be loaded on the host operating system.

> On Windows and Linux workers without SELinux support, this field and any SELinux feature gates described

Reviewer: ```suggestion
On Windows and Linux worker nodes without SELinux support, this field and any SELinux feature gates described
```

Kubernetes v1.27 introduced an early limited form of this behavior that was only applicable
to volumes (and PersistentVolumeClaims) using the `ReadWriteOncePod` access mode.

> Kubernetes v1.33 introduced `SELinuxChangePolicy` and `SELinuxMount`

Reviewer: ```suggestion
Kubernetes v1.33 promotes `SELinuxChangePolicy` and `SELinuxMount`
```

[feature gates](/docs/reference/command-line-tools-reference/feature-gates/)
as beta to widen that performance improvement to other kinds of PersistentVolumeClaims,

> as explained in detail below. While beta, `SELinuxMount` is still disabled by default.

Reviewer: ```suggestion
as explained in detail below. While in beta, `SELinuxMount` is still disabled by default.
```

    The CSI driver must announce that it supports mounting with `-o context` by setting
    `spec.seLinuxMount: true` in its CSIDriver instance.

When any of these conditions is not met, SELinux relabelling happens another way: the container
runtime  recursively changes the SELinux label for all inodes (files and directories)
in the volume. Calling out explicitly, this applies to Kubernetes ephemeral volumes like

> `secrets`, `configMaps` and `projected`, and all volumes whose CSIDriver instance does not

Reviewer: ```suggestion
`secret`, `configMap` and `projected`, and all volumes whose CSIDriver instance does not
```

A cluster admin can use this information to identify pods affected by the planning change and
proactively opt-out Pods from the optimization (i.e. set `spec.securityContext.seLinuxChangePolicy: Recursive`).

{{< warning >}}
We strongly recommend clusters that use SELinux to enable this controller and make sure that

> `selinux_warning_controller_selinux_volume_conflict` metric does not report any conflicts before enabling `SELinuxMount`

Reviewer: are you sure that this metrics will report anything before `SELinuxMount` is enabled?

Author: Yes. That's why the whole new controller is introduced. Just to expose metrics + events for Pods that would break *before* `SELinuxMount` is enabled.

Reviewer: Got it. Thanks.

* `SELinuxChangePolicy`: enables `spec.securityContext.seLinuxChangePolicy` field in Pod and related SELinuxWarningController
  in kube-controller-manager. This feature can be used before enabling `SELinuxMount` to check Pods running on a cluster,
  and to pro-actively opt-out Pods from the optimization.
  This feature gate requires `SELinuxMountReadWriteOncePod` enabled. It is beta and enabled by default in 1.33.
* `SELinuxMount` enables the optimization for all eligible volumes. Since it can break existing workloads, we recommend
  enabling `SELinuxChangePolicy` feature gate + SELinuxWarningController first to check the impact of the change.

>   This feature gate requires `SELinuxMountReadWriteOncePod` and `SELinuxChangePolicy` enabled. It is beta, but disabled

Reviewer: Which gate is "It" referring to?
`SELinuxMountReadWriteOncePod` is enabled by default right?

Author: "It" refers to SELinuxMount.
